app.js

```diff
@@ -2,17 +2,54 @@ require('dotenv').config();
 
 const express = require('express');
 const hbs = require('hbs');
+const SpotifyWebApi = require('spotify-web-api-node');
 
-// require spotify-web-api-node package here:
 
 const app = express();
 
 app.set('view engine', 'hbs');
 app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
 
-// setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+    clientId: process.env.CLIENT_ID,
+    clientSecret: process.env.CLIENT_SECRET
+  });
+
+  spotifyApi
+  .clientCredentialsGrant()
+  .then(data => spotifyApi.setAccessToken(data.body['access_token']))
+  .catch(error => console.log('Something went wrong when retrieving an access token', error));
 
 // Our routes go here:
 
+app.get('/', (req, res) => {
+    res.render('home-page');
+})
+
+app.get('/artist-search', (req, res) => {
+    // console.log(req.query.artist);
+    //console.log('hey');
+    spotifyApi
+  .searchArtists(req.query.artist)
+  .then(data => {
+    //console.log('The received data from the API: ', data.body);
+    const artistsResult = data.body.artists.items;
+    //console.log(artistsResult);
+    console.log('hello');
+    const imageResult = artistsResult[0].images;
+    console.log(imageResult);
+  // ----> 'HERE WHAT WE WANT TO DO AFTER RECEIVING THE DATA FROM THE API'
+
+    // const filteredArtists = onlyArtist.filter(function (artist) {
+    //   return data.onlyArtist.includes(req.query.artist);
+    // })
+    // console.log(filteredArtists);
+    res.render('artist-search-results', {artistsResult, imageResult})
```

Comment:
Here you could the search result data (all the artists) to the hbs `{ artistResult: data.body.artists.items }`